src/lib/Service/Event/SuggestionService.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Search\Service\Event;
+
+use Ibexa\Contracts\Search\Event\AfterSuggestionEvent;
+use Ibexa\Contracts\Search\Event\BeforeSuggestionEvent;
+use Ibexa\Contracts\Search\Model\Suggestion\SuggestionCollection;
+use Ibexa\Contracts\Search\Service\Decorator\SuggestionServiceDecorator;
+use Ibexa\Contracts\Search\Service\SuggestionServiceInterface;
+use Ibexa\Search\Model\SuggestionQuery;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+final class SuggestionService extends SuggestionServiceDecorator
+{
+    private EventDispatcherInterface $eventDispatcher;
+
+    public function __construct(
+        SuggestionServiceInterface $innerService,
+        EventDispatcherInterface $eventDispatcher
+    ) {
+        parent::__construct($innerService);
+
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function suggest(SuggestionQuery $query): SuggestionCollection
+    {
+        $beforeEvent = $this->eventDispatcher->dispatch(
+            new BeforeSuggestionEvent(
+                $query,
+                new SuggestionCollection()
```

Comment:
I think this collection should not be set. It should be left unset until someone handling the event fills it. Otherwise it's a logic error (if propagation is stopped, yet no results are set).